index.html

```diff
@@ -54,6 +54,7 @@
 <body>
   <noscript>You need to enable JavaScript to run this app.</noscript>
   <div id="root"></div>
+  <div id="modal"></div>
```

Comment:
I just wanna ask, what's the reason for adding the modal here ?

Reply:
Actually I am creating a portal, means modal will get rendered into this `#modal` doesn't matter which nested component implement the modal. This is an efficient way to create modals, for example our modal has `z-index: 20` if we the component who's nested like more than 20 level deep then our modal will not be on the top. So to solve this we use portal to render modal on the root level element

Read more on this:
https://react.dev/reference/react-dom/createPortal#rendering-a-modal-dialog-with-a-portal

Reply:
Oh , now i get it ! 
Thanks for letting me know about this.

**Amazing work,**
This looks good to me, will be merged soon 🎉.